PennMobile/src/main/java/com/pennapps/labs/pennmobile/adapters/LaundryMachineAdapter.java

```diff
@@ -0,0 +1,192 @@
+                            .append(machine.number);
+                    Snackbar snackbar = Snackbar.make(buttonView,stringBuilder , Snackbar.LENGTH_SHORT)
+                            .setAction(activity.getString(R.string.laundry_undo), new View.OnClickListener(){
+                                @Override
+                                public void onClick(View view){
+                                    button.setChecked(false);
+                                    alarmManager.cancel(alarmIntent);
+                                    alarmIntent.cancel();
+                                }
+                            });
+                    View subView = snackbar.getView();
+                    TextView snackTextView = (TextView) subView.findViewById(android.support.design.R.id.snackbar_text);
+                    snackTextView.setTextColor(activity.getResources().getColor(R.color.white));
+                    snackbar.show();
+                } else {
+                    final PendingIntent alarmIntent = PendingIntent.getBroadcast(getContext(), 50 * laundry.hall_no + machine.number,
+                            intent, PendingIntent.FLAG_NO_CREATE);
+                    if (alarmIntent != null) {
+                        alarmManager.cancel(alarmIntent);
+                        alarmIntent.cancel();
+                    }
+                    StringBuilder stringBuilder = new StringBuilder();
+                    stringBuilder.append(activity.getString(R.string.laundry_notification_snackbar_off))
+                            .append(" ").append(machine.machine_type).append(" ")
+                            .append(machine.number);
+                    Snackbar snackbar = Snackbar.make(buttonView,stringBuilder, Snackbar.LENGTH_SHORT)
+                            .setAction(activity.getString(R.string.laundry_undo), new View.OnClickListener(){
+                                @Override
+                                public void onClick(View view){
+                                    button.setChecked(true);
+                                    PendingIntent alarmIntent = PendingIntent.getBroadcast(getContext(), 50 * laundry.hall_no + machine.number,
+                                            intent, PendingIntent.FLAG_CANCEL_CURRENT);
+                                    alarmManager.set(AlarmManager.ELAPSED_REALTIME_WAKEUP, SystemClock.elapsedRealtime() + 3000, alarmIntent);
+                                }
+                            });
+                    View subView = snackbar.getView();
+                    TextView snackTextView = (TextView) subView.findViewById(android.support.design.R.id.snackbar_text);
+                    snackTextView.setTextColor(activity.getResources().getColor(R.color.white));
+                    snackbar.show();
+                }
+            }
+        });
+    }
+
+    @Override
+    public int getCount(){
+        return machines.size() + 1;
```

Comment:
Because we have the summary as the first item of the list, so there are list.size+1 things on the list.